### PR TITLE
require confirmation when the user switches post types

### PR DESCRIPTION
**Pull request by aliceriot**
#### Pre-Flight checklist

- [x] Screenshots and design review for any changes that affect layout or styling
  - [x] tag @pdpinch for review  
- [x] Testing
  - [x] Code is tested
  - [x] Changes have been manually tested

#### What are the relevant tickets?

part of #1583 

#### What's this PR do?

This adds a confirmation dialog if the user tries to remove the post content from their post (click the little 'x') and they have already added some input. If they haven't made any input (typed in a url, text post, or article) we don't put up the confirmation.

I just did my best guess as to what the copy should be.

#### How should this be manually tested?

Go to the post creation page. If you click the post-type buttons you should be able to click the 'x' and it should clear away the input straightaway. If you click a post-type button and then add some input, it should confirm before it clears it away. Cancelling with the dialog should not clear away your input.

#### Screenshots (if appropriate)
clearpostconetnttt

### CODE REVIEW

File: static/js/components/CreatePostForm.js [hunk 1]
```diff
@@ -94,11 +96,21 @@ export default class CreatePostForm extends React.Component<Props> {
     )
   }
 
+  clearInputClickHandler = (e: Event) => {
+    e.preventDefault()
```
[the remark this answers is not in the document]
Comment (aliceriot) [reply static/js/components/CreatePostForm.js]: yeah I can change that, I sometimes shy away from it to avoid allocating an extra function object but it's really not that much overhead

[the remark this answers is not in the document]
Comment (noisecapella) [reply static/js/components/CreatePostForm.js]: Sure, up to you

**Comment by aliceriot** [comment]
@noisecapella ok so I added a test for that function, added in the `prevetnDefaultAndInvoke` function. I also noticed a few edge cases where the post form wasn't being cleared in the way that it should be when switching between channels (which allow different kinds of posts) and when clearing the selected post type. I think it should all be handled now, but if you want to test it again and make sure that it behaves as  you expect when switching between channels, adding and removing the post type, adding a title, etc.
